tests/4.5/target_update/test_target_update_from.c

```diff
@@ -1,73 +1,66 @@
-//===---- test_target_update_to.c - check the to data motion clause of target update -===//
+//===---- test_target_update_from.c -------------------------------------------===//
 // 
 // OpenMP API Version 4.5 Nov 2015
-// 
-//===----------------------------------------------------------------------===//
+//
+// This test checks the target update motion clause 'from' by mapping an array 
+// to the device with map-type 'to', changing the values of array on the device,
+// and finally using the update 'from' motion clause to assign the value of the 
+// list item. Back on the host, measures are taken to ensure the value was properly
+// updated.  
+//===--------------------------------------------------------------------------===//
 
 #include <omp.h>
 #include <stdio.h>
+#include "ompvv.h"
 
 #define N 100
 int a[N];
 int b[N];
 int c[N];
 
-void update_b(){
-    int i;
-    for (i = 0; i < N; i++) {
-      b[i] = b[i] * 2; 
-    }
-}
-  
+
 // Test for OpenMP 4.5 target update with to
 int main() {
-  int errors= 0, i = 0, isHost = -1, isOffloading = 0, change_flag=0;
+  int errors = 0, i = 0, change_flag = 0;
 
   for (i = 0; i < N; i++) {
     a[i] = 10;
     b[i] = 2; 
   }
 
   // We test for offloading
-#pragma omp target map(from: isOffloading)
-  {
-    isOffloading = !omp_is_initial_device();
-  }
+  int is_offloading;
+  OMPVV_TEST_AND_SET_OFFLOADING(is_offloading);
 
-#pragma omp target data map(to: a[:N], b[:N]) map(tofrom: isHost)
+
+#pragma omp target data map(to: a[:N], b[:N]) 
 {
   #pragma omp target
   {
-        isHost = omp_is_initial_device();
-        int j = 0;
-        for (j = 0; j < N; j++) {
-          b[j] = (a[j] + b[j]);//b=12 
-        }
+    int j = 0;
+    for (j = 0; j < N; j++) {
+      b[j] = (a[j] + b[j]);//b=12 
+    }
   } // end target
 
   #pragma omp target update from(b[:N]) //update b=12 on host 
 
```

Comment:
I don't see the value of modifying and testing array c. We could very well end device and date env at 47 and check value of b.

Reply:
I agree, should I go ahead and remove target section where c is dealt with (lines 48-53), as well as remove lines 59-60 where c is checked?